enterprise/cmd/frontend/internal/authz/resolvers/resolver.go

```diff
@@ -544,6 +552,91 @@ func (r *permissionsInfoResolver) Unrestricted() bool {
 	return r.unrestricted
 }
 
+var permissionsInfoRepositoryConnectionMaxPageSize = 100
+
+var permissionsInfoRepositoryConnectionOptions = &graphqlutil.ConnectionResolverOptions{
+	OrderBy:     database.OrderBy{{Field: "repo.name"}},
+	Ascending:   true,
+	MaxPageSize: &permissionsInfoRepositoryConnectionMaxPageSize,
+}
+
+func (r *permissionsInfoResolver) Repositories(ctx context.Context, args graphqlbackend.PermissionsInfoRepositoriesArgs) (*graphqlutil.ConnectionResolver[graphqlbackend.PermissionsInfoRepositoryResolver], error) {
+	if r.userID == 0 {
+		return nil, nil
+	}
+
+	query := ""
+	if args.Query != nil {
+		query = *args.Query
+	}
+
+	connectionStore := &permissionsInfoRepositoriesStore{
+		userID: r.userID,
+		db:     r.db,
+		ossDB:  r.ossDB,
+		query:  query,
+	}
+
+	return graphqlutil.NewConnectionResolver[graphqlbackend.PermissionsInfoRepositoryResolver](connectionStore, &args.ConnectionResolverArgs, permissionsInfoRepositoryConnectionOptions)
+}
+
+type permissionsInfoRepositoriesStore struct {
```

Comment:
Can we use less complex naming here? E.g.
```suggestion
type repositoryPermissionsStore struct {
```